lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        # Retrieves job from redis.
+        #
+        # @return [Sidekiq::Throttled::UnitOfWork, nil]
+        def retrieve_work
```

Comment:
Is it possible to reuse `Sidekiq::Throttled::Patches::BasicFetch.retrieve_work` instead of copy paste?

Reply:
I think we can extract that into `AnyFetch` module indeed:

``` ruby
module Patches
  module ThrottledRetriever
    def retrieve_work
      ...
    end
  end

  module BasicFetch
    def self.included(base)
      base.prepend(ThrottledRetriever)
    end

    ...
  end

  module SuperFetch
    def self.included(base)
      base.prepend(ThrottledRetriever)
    end

    ...
  end
end
```

Reply:
https://github.com/ixti/sidekiq-throttled/pull/177

Reply:
@mnovelo

Reply:
Sounds good! I just totally forgot to go back and refactor, so I'll rebase onto #177